### PR TITLE
chore(docs): create architecture page

### PR DESCRIPTION
**Pull request by sfirke**
### SUMMARY

- Adds an "architecture" page summarizing the components of a Superset installation.  A visual diagram should eventually go here too.  Please feel free to edit/add/cut.
- Reorders other installation pages to follow this one
- Copyediting of Configuring Superset page

### CODE REVIEW

File: docs/docs/configuration/configuring-superset.mdx [hunk 1]
```diff
@@ -87,23 +87,15 @@ can be altered in your local `superset_config.py`. Administrators will want to r
...
 for more information on how to configure it.
 
-You'll want to change:
-
-- `SECRET_KEY`: to a long random string
-- `SQLALCHEMY_DATABASE_URI`: that by default points to sqlite database located at
-  ~/.superset/superset.db
-
-```
-WTF_CSRF_EXEMPT_LIST = [‘’]
```
Comment (sfirke) [inline docs/docs/configuration/configuring-superset.mdx:105]: This line lingered accidentally after other content was deleted.  The same variable already appears higher above on this page.

**Comment by sfirke** [comment]
This is ready for review.

File: docs/docs/installation/architecture.mdx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import useBaseUrl from "@docusaurus/useBaseUrl";
+
+# Architecture
+
+This page is meant to give new administrators an understanding of Superset's components.
+
+## Components
+
+A Superset installation is made up of these components:
+1. The Superset application itself
```
Comment (mistercrunch) [inline docs/docs/installation/architecture.mdx:17]: Should we call it "the Superset [python] backend"?

Comment (mistercrunch) [reply docs/docs/installation/architecture.mdx:17]: Should we call out "The Superset frontend" - a collection of React application bundled with webpack and using antd, ...

File: docs/docs/installation/architecture.mdx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+2. A metadata database to store Superset's data about users, charts, dashboards, etc.
+3. A caching layer (optional, but necessary for some features)
```
Comment (mistercrunch) [inline docs/docs/installation/architecture.mdx:19]: this is called out twice in "components" and "optional component", i think it's a great idea to seperate the section, but let's have each component on just one side

**Comment by mistercrunch** [comment]
hey! - I'm hoping it doesn't come across as a hostile takeover of this PR, but I threw your pages into GPT and added some comments/input in the form of bullet, and this is what came out:
-----
# **Architecture**

This documentation outlines the architecture of Apache Superset, with a primary focus on the backend components before introducing the frontend. This organization provides a thorough understanding of how Superset operates from data handling to user interface.

## **Superset Backend**

The backend of Superset is composed of several critical components designed to manage data, execute tasks, and maintain the overall functionality of the system.

### **Core Backend Components**

**Web Application [Python/Flask]**:

- **Description**: Serves static assets and handles API requests from the Superset frontend.
- **Role**: Acts as the primary communication hub for frontend interactions and immediate query processing.

**Metadata Database [Required]**:

- **Description**: Stores all of Superset’s essential assets such as dashboards, charts, user configurations, and logs.
- **Supported Technologies**: PostgreSQL or MySQL (recommended), other SQLAlchemy-supported OLTP databases

### **Optional Backend Components**

**Asynchronous Workers [Pytjhon/Celery]**:

- **Description**: Manages tasks that are too long or intense for a typical web request cycle.
- **Enabled Features**:
    - Asynchronous query executions in SQL Lab.
    - Scheduled generation of alerts and reports.
    - Creation of dashboard thumbnails.
- **Dependencies**: Requires a message queue (e.g., Redis, RabbitMQ).

**Caching Layer**:

- **Description**: Enhances performance by caching query results and frequently accessed data.
- **Technologies**: Primarily Redis, with support for other caching systems.
- **Enabled Features**:
    - Accelerated access to repeated queries.
    - Improved responsiveness of the application.

### **Logging Interfaces**

- **Standard Output and Error Logs**: Essential for debugging and monitoring application health.
- **StatsD/Metrics Collection**: Enables real-time aggregation of performance metrics.
- **Analytics Logging**: Rich structured logs that provides insights into user behaviors and application usage patterns. Typically sent to a stream to land into a data warehouse

### **Other Common Infrastructure components**:

- **Load Balancers/API Gateway**: Distributes incoming traffic across multiple servers to enhance availability and manage traffic peaks.
- **Observability/Alerting**: Provides monitoring, error tracking, and real-time alerts to maintain performance and uptime.
- **WSGI Server (e.g., Gunicorn in async mode)**:
- **Database Drivers**: Enables communication between Superset and its databases, crucial for operational data querying and management.
- **Orchestration (e.g., Kubernetes)**: Automates deployment, scaling, and management of containerized applications, ensuring robust service availability.
- **Additional Security Measures**: Implements network security, data encryption, and access controls to safeguard data and comply with regulations.

## **Superset Frontend**

The frontend of Superset is a sophisticated web client built using modern web technologies to facilitate interactive data visualization.

**Core Technologies**:

- **React**: Forms the foundation of the frontend, offering a responsive and dynamic user interface.
- **antd**: Utilized for designing the visual components and layout of the UI, providing a consistent and professional aesthetic.
- **Plugin Architecture**: Allows for the extension of visualization capabilities through custom plugins, enhancing the flexibility and functionality of visual data representation.

**Functionality**:

- Communicates with the backend via the Superset API, enabling users to manage and visualize data efficiently.
- Supports extensive customization and extension through community-developed plugins and themes.

**Comment by rusackas** [comment]
Github needs the "face melt" emoji as a reaction. 

Is it possible to "land and expand" here? My trust issues with GPT aside (did it really say "Pytjhon?") I wonder if we can merge a first iteration, then divide/expand/remove/elaborate as needed from there, rather than go deep here. We can feed it to GPT as we go for consolidation/clarification/organization.

**Comment by mistercrunch** [comment]
I did a fair amount of prompt-inputs and edits to get to that, but main thing is the structure of the docs (backend/frontend) and mentioning technologies used (and technology choices) in different areas

File: docs/docs/installation/architecture.mdx (hunk 2 above)
Comment (rusackas) [inline docs/docs/installation/architecture.mdx:17]:
```suggestion
1. **The Superset application itself**. This includes the Python (Flask) backend application, API layer, and the React frontend, built via Webpack, and static assets needed for the application to work.
```

File: docs/docs/installation/architecture.mdx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+# Architecture
+
+This page is meant to give new administrators an understanding of Superset's components.
+
+## Components
+
+A Superset installation is made up of these components:
+1. The Superset application itself
+2. A metadata database to store Superset's data about users, charts, dashboards, etc.
```
Comment (rusackas) [inline docs/docs/installation/architecture.mdx:18]:
```suggestion
2. **A metadata database**. This stores Superset's data about users, charts, dashboards, etc.
```

File: docs/docs/installation/architecture.mdx (hunk 3 above)
Comment (rusackas) [inline docs/docs/installation/architecture.mdx:19]:
```suggestion
3. **A caching layer**. Optional, but necessary for some features. There is more information about this below.
```

File: docs/docs/installation/architecture.mdx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+However, installing from PyPI only creates the application itself. Users installing from PyPI will need to configure a caching layer, worker, and beat on their own if they wish to enable the above features. Configuration of those components for a PyPI install is not currently covered in this documentation.
+
+Here are further details on each component.
+
+### The Superset Application
+
+This is the core application. Superset operates like this:
+- A user visits a chart or dashboard
+- That triggers a SQL query to the data warehouse holding the underlying dataset
+- The resulting data is served up in a data visualization
```
Comment (rusackas) [inline docs/docs/installation/architecture.mdx:41]:
```suggestion
- The resulting data is served up in a data visualization
- The Superset application is comprised of the Python (Flask) backend application (server), API layer, and the React frontend, built via Webpack, and static assets needed for the application to work
```

File: docs/docs/installation/architecture.mdx (hunk 2 above)
Comment (rusackas) [reply docs/docs/installation/architecture.mdx:17]: I mention this in a suggestion below in the description of the Superset application.

**Review by rusackas — APPROVED** [review]
We can circle back and make this list link to the descriptions of each component below, but let's just merge this thing and dial it in from there.

**Comment by rusackas** [comment]
From here, we can update the installation page to provide a little table (or similar) about how (or if) each installation method installs these components.

**Pull request merged**